async_service/function_service/fastapi_generator.py

```diff
@@ -0,0 +1,87 @@
+from async_service.function_service.utils import (
+    create_pydantic_model_from_function_signature,
+    get_functions_dict_with_input_signatures
+)
+from pydantic import BaseModel
+from fastapi import FastAPI
+from async_service import (
+    InputMessage,
+    Processor,
+    WorkerConfig,
+    InputConfig,
+    OutputConfig,
+)
+import json
+import uuid
+from typing import Callable, Dict
+
+INTERNAL_FUNCTION_NAME = "internal_func_name"
+
+
+async def send_request_to_queue(
+    request_id: str, input: BaseModel, output_config: OutputConfig
+):
+    output_publisher = output_config.to_output()
+    
+    my_dict = dict(input._iter(to_dict=False))
+    my_dict[INTERNAL_FUNCTION_NAME] = input.__class__.__name__
+    input_message = InputMessage(request_id=request_id, body=my_dict)
+    
+    await output_publisher.publish_output_message(
+        request_id=request_id,
+        serialized_output_message=json.dumps(input_message.dict()).encode(),
+    )
+
+
+def wrapper_func(func, name: str, output_config: OutputConfig):
+    async def wrapper(input: create_pydantic_model_from_function_signature(func, name)):
+        request_id = str(uuid.uuid4())
+        await send_request_to_queue(request_id, input, output_config)
+        return request_id
+
+    return wrapper
+
+
+def create_async_server_for_functions(
+    funct_dict: dict[str, callable], output_config: OutputConfig
+):
+    app = FastAPI()
+    app.add_api_route(
+        "/function-schemas",
+        lambda : get_functions_dict_with_input_signatures(funct_dict),
+        methods=["GET"],
+    )
+
+    for name, func in funct_dict.items():
+        app.add_api_route(
+            f"/{name.lower()}",
+            wrapper_func(func, name, output_config),
+            methods=["POST"],
+        )
+
+    return app
+
+
+def create_worker_for_functions(
```

Comment:
Can we add a usage example? I am missing the high level user interface
I was expecting it would be something like `register_function(func: Callable[..., Any], name: Optional[str] = None)` on a class


```
fn_as = FunctionAsyncService(...)
fn_as.register_function(...)
fn_as.register_function(...)
fn_as.register_function(...)
fn_as.register_function(...)
fn_as.run(...) or # app = fn_as.asgi_app() 
```

Reply:
fn_as = FunctionAsyncService(...)
fn_as.register_function(...)
fn_as.register_function(...)
fn_as.register_function(...)
fn_as.register_function(...)
fn_as.run(...) or # app = fn_as.asgi_app() 

we could do that ideally, creating_worker_app and create_server_app will be different and their config would be passed in the function params only instead of class init.

So Idk if this class will be adding much value in that case

Reply:
```python
def foo():
    ...


def init():
    ...


functions = {"foo": foo}

input_config = SQSInputConfig()
output_config = NATSOutputConfig()


async_function_deployment = AsyncFunctionDeployment(
    init_function=init,
    functions=functions,
    worker_config=WorkerConfig(
        input_config=input_config,
        output_config=output_config,
    ),
)

some_app = async_function_deployment.build_some_app()  # REST
worker_app = async_function_deployment.build_worker_app()  # Worker
```